### DISCUSSION TITLE
[BUG] FastAPI does not use Pydantic model's configs

### DISCUSSION THREAD

**Discussion by RB387** [fastapi/fastapi#12503] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel, ConfigDict


class Missing:
    ...

MISSING = Missing()



class Model(BaseModel):
    model_config = ConfigDict(arbitrary_types_allowed=True)

    value: str | Missing = MISSING


app = FastAPI()


@app.post("/")
async def read_root(a: Model) -> bool:
    return True
```

### Description

* Request `﻿curl -X POST 0.0.0.0:8000 -d '{}' -H "Content-Type: application/json"` 
Returns an error: ﻿`pydantic.errors.PydanticSchemaGenerationError: Unable to generate pydantic-core schema for <class '__main__.Missing'>. Set arbitrary_types_allowed=True﻿in the model_config to ignore this error or implementget_pydantic_core_schema﻿ on your type to fully support it.`
* FastAPI 0.112.0 functions correctly (app responds with 200 status code).
* The issue may be due to this line in the code: https://github.com/fastapi/fastapi/blob/45822d31f265861dff27e9d2a935c87aafb101b3/fastapi/_compat.py#L110 where the Model config is not being passed to the adapter.

### Operating System

macOS

### FastAPI Version

0.115.1

### Pydantic Version

2.9.2

### Python Version

Python 3.12.0

**Comment by RB387** [comment]
https://github.com/fastapi/fastapi/pull/12504

**Comment by RB387** [comment]
Could someone please take a look at this bug?

**Reply by rmawatson** [reply]
Found the same problem [here](https://github.com/fastapi/fastapi/discussions/13690). This used to work but a change between 0.112.4 and 0.113.0 seems to have broken it, did you find a solution?

**Reply by rmawatson** [reply, marked as answer]
fixed by https://github.com/fastapi/fastapi/pull/13694

**Reply by YuriiMotov** [reply]
In current version FastAPI fails to generate openapi schema for the code example above
